moveit_core/kinematic_constraints/src/kinematic_constraint.cpp

```diff
@@ -620,9 +620,9 @@ ConstraintEvaluationResult OrientationConstraint::decide(const moveit::core::Rob
     xyz = diff.linear().eulerAngles(0, 1, 2);  // 0,1,2 corresponds to XYZ, the convention used in sampling constraints
   }
 
-  xyz(0) = std::min(fabs(xyz(0)), boost::math::constants::pi<double>() - fabs(xyz(0)));
-  xyz(1) = std::min(fabs(xyz(1)), boost::math::constants::pi<double>() - fabs(xyz(1)));
-  xyz(2) = std::min(fabs(xyz(2)), boost::math::constants::pi<double>() - fabs(xyz(2)));
+  xyz(0) = std::min(fabs(xyz(0)), boost::math::constants::two_pi<double>() - fabs(xyz(0)));
+  xyz(1) = std::min(fabs(xyz(1)), boost::math::constants::two_pi<double>() - fabs(xyz(1)));
+  xyz(2) = std::min(fabs(xyz(2)), boost::math::constants::two_pi<double>() - fabs(xyz(2)));
```

Comment:
Given that the values we get from [eulerAngles](https://eigen.tuxfamily.org/dox/group__Geometry__Module.html#ga17994d2e81b723295f5bc3b1f862ed3b) are within the following ranges `[0:pi]x[-pi:pi]x[-pi:pi]` and this [graph](https://www.desmos.com/calculator/wru9zmuvic) I think we could use the abs value of the vector directly

Reply:
Thanks for the review!  I noticed that as well, but assumed the original author wasn't sure that eulerAngles would always return values in that range in the future if its implementation changed for some reason.  If you're confident that eulerAngles will never change its implementation then it makes sense to simplify here.

Reply:
Agree with Stuart - better not count on a consistent implementation